spec/features/auth_spec.rb

```diff
@@ -16,4 +16,26 @@
     expect(page).to have_content("user@example.com")
   end
 
+  scenario "Users can register and are automatically logged in" do
+    create_user email: "user@example.com"
+
+    visit root_path
+    click_link "Register"
+
+    within(".registration-form") { click_on "Register" }
+
+    fill_in "Name", with: "Mickey Mouse"
+    fill_in "Email", with: "mickey@mouse.com"
+    fill_in "Password", with: "password"
+    fill_in "Confirm", with: "password"
+
+    within(".registration-form") { click_on "Register" }
+
+    expect(page).to have_content("Welcome mickey@mouse.com")
+
+
+
+  end
```

Comment:
Awesome and correct answer. An improvement would be to not add all of this unnecessary whitespace.